pkg/controllers/disruption/consolidation.go

```diff
@@ -156,18 +163,16 @@ func (c *consolidation) computeConsolidation(ctx context.Context, candidates ...
 	if err != nil {
 		return Command{}, fmt.Errorf("getting offering price from candidate node, %w", err)
 	}
-	results.NewNodeClaims[0].InstanceTypeOptions = filterByPrice(results.NewNodeClaims[0].InstanceTypeOptions, results.NewNodeClaims[0].Requirements, candidatePrice)
-	if len(results.NewNodeClaims[0].InstanceTypeOptions) == 0 {
+
+	instanceTypesOptionsFilteredByPrice := filterByPrice(results.NewNodeClaims[0].InstanceTypeOptions, results.NewNodeClaims[0].Requirements, candidatePrice)
+	if len(instanceTypesOptionsFilteredByPrice) == 0 {
```

Comment:
Can you add a comment here on why we don't use the instanceTypesOptionsFilteredByPrice as a parameter to `c.computeSpotToSpotConsolidation()`?  I understand why we're not, but I think it would be worth a comment here for future readers.

I'd even change the ordering of the code to put this block below the spot-to-spot block so that it's more clear that it only pertains to non-spot-to-spot consolidation.

https://github.com/kubernetes-sigs/karpenter/pull/897#discussion_r1436227898

Reply:
Added.